### ISSUE TITLE
Different output based on Windows HiDPI scaling

### ISSUE THREAD

**Issue by WillemJann** [wkhtmltopdf/wkhtmltopdf#2191]
I am using wkhtmltopdf 0.12.2.1 (with patched qt) on Windows 8.1 with the following command:

```
wkhtmltopdf -B 0 -L 0 -R 0 -T 0 -O Landscape --disable-smart-shrinking http://localhost:1922/page.html output.pdf
```

The page is styled with the following CSS:

``` css
body {
    margin: 0;
    padding: 0;
    background-color: #0269da;
    width: 297mm;
    height: 210mm;
}
```

This produces the following output on my laptop:

![test-scale-2](https://cloud.githubusercontent.com/assets/3678597/6024953/37e2e68e-abcf-11e4-887a-3ef0bf985b6e.jpg)

However, it is expected that the output PDF would fill the entire page. More research on this bug turned out that the issue is caused by the Windows scaling setting. I am using a laptop with a HiDPI screen and have set the Windows scaling to 200%.

When I change the Windows scaling setting to 100%, suprisingly the correct output is generated:

![test-scale-1](https://cloud.githubusercontent.com/assets/3678597/6025009/a215fb68-abcf-11e4-82d6-d4f11ef31e49.jpg)

It is an unwanted side-effect that the output depends on the scaling setting of Windows. Are there any possibilities to generate the last PDF (expected outcome), even when scaling is set to 200%?

**Comment by ashkulz** [comment]
Does using an explict `--dpi` help?

**Comment by WillemJann** [comment]
Unfortunately no, when I use for example `--dpi 300` the output is the same. So the issue seems directly related to the Windows scaling setting (maybe due the underlying Qt libary?).

**Comment by ashkulz** [comment]
I suspect it is due to `qt_defaultDpi` returning the actual Windows DPI.

**Comment by WillemJann** [comment]
But with `--dpi 300` the problem still exist (output is the same as in the above post). That means that on machines with the Windows scaling setting != 100% the output is not as expected.

Maybe you have ideas to make the conversion independent of the Windows DPI setting (either using a wkhtmltopdf setting or with custom CSS on the actual webpage)?

**Comment by rjurado01** [comment]
+1

**Comment by barsch** [comment]
+1

**Comment by GitteHubert** [comment]
Seems to be a duplicate of #1641

**Comment by rainabba** [comment]
I've encountered this a few times now. Each time, it's been because I connected to the server running my app, using my laptop (Surface Book or Sager with 4K display) running in 200% DPI mode and I could fix it by using `zoom` which is how I came to realize the issue. If I logged out of that session, connected with a machine using 100% DPI, then started my app, the RDP session was running 100% DPI and so I could use `zoom: 1` (where I needed `zoom: 2.0` otherwise.)

I wonder if WKHTML2PDF could detect the desktop DPI through a Windows API, then compensate (offset zoom) appropriately?

Basically, `useZoom = windowsDpiPercent / requestedZoom` so that `1 = 2.0 / 1.0` where 2.0 is the 200% DPI.
